server/sse.go

```
@@ -136,30 +148,31 @@ func (s *SSEServer) handleSSE(w http.ResponseWriter, r *http.Request) {
-		done:       make(chan struct{}),
-		eventQueue: make(chan string, 100), // Buffer for events
+		writer:              w,
+		flusher:             flusher,
+		done:                make(chan struct{}),
+		eventQueue:          make(chan string, 100), // Buffer for events
+		sessionID:           sessionID,
+		notificationChannel: make(chan mcp.JSONRPCNotification, 100),
 	}
 
 	s.sessions.Store(sessionID, session)
+	s.server.RegisterSession(session)
```

golangci-lint (1.62.2)

Error return value of `s.server.RegisterSession` is not checked

(errcheck)